src/System.IO.FileSystem.DriveInfo/tests/DriveInfo.Windows.Tests.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Xunit;
+using System.Text;
+
+namespace System.IO.FileSystem.DriveInfoTests
+{
+    public class DriveInfoTests
+    {
+        [Fact]
+        public void TestConstructor()
+        {
+            string[] invalidInput = { ":", "://", @":\", ":/", @":\\", "Az", "1", "a1", "\\share", "\\", "c ", string.Empty, " c" };
+            string[] variableInput = { "{0}", "{0}", "{0}:", "{0}:", @"{0}:\", "{0}:\\", "{0}://" };
+
+            // Test Invalid input
+            foreach (var input in invalidInput)
+            {
+                Assert.Throws<ArgumentException>(() => { new DriveInfo(input); });
+            }
+
+            // Test Null
+            Assert.Throws<ArgumentNullException>(() => { new DriveInfo(null); });
+
+            // Test Valid DriveLetter
+            var validDriveLetter = GetValidDriveLettersOnMachine().First();
+            foreach (var input in variableInput)
+            {
+                string name = string.Format(input, validDriveLetter);
+                DriveInfo dInfo = new DriveInfo(name);
+                Assert.Equal(string.Format(@"{0}:\", validDriveLetter), dInfo.Name);
+            }
+        }
+        
+        [Fact]
+        public void TestGetDrives()
+        {
+            var validExpectedDrives = GetValidDriveLettersOnMachine();
+            var validActualDrives = DriveInfo.GetDrives();
+
+            // Test count
+            Assert.Equal(validExpectedDrives.Count(), validActualDrives.Count());
+
+            for (int i = 0; i < validActualDrives.Count(); i++)
+            {
+                // Test if the driveletter is correct
+                Assert.True(validExpectedDrives.Contains(validActualDrives[i].Name[0]));
```

Comment:
`Assert.Contains` is shorter and provides a better error message if the item is not contained in the collection.